### ISSUE TITLE
Accessing 'IgnoreFilesIndex' during processing 'filetypes'.

### ISSUE THREAD

**Issue by mayfma9** [hsz/idea-gitignore#473]
### Prerequisites

* [x] Plugin is in the latest version
* [x] Issue was not reported yet
* [x] Stack trace (if provided) contains `mobi.hsz.idea.gitignore` package name

### Description

Restarted Android Studio before an update. Android Studio reopened the project I was working on and suddenly almost every codeline had an error that begins with "Cannot resolve... ". The Event Log shown the following:
```
02:31 PM	Gradle sync started

02:31 PM	Plugin Error: .ignore threw an uncaught EmptyThrowable. Disable Plugin

02:31 PM	EmptyThrowable: Accessing 'IgnoreFilesIndex' during processing 'filetypes'. Nested different indices processing may cause deadlock

02:31 PM	Gradle sync completed
```

### Additional Information

Projects compile and run on debug mode normally. However, as almost everything is red, it's confusing to the point of annoyance.

### Stack Trace

```
Accessing 'IgnoreFilesIndex' during processing 'filetypes'. Nested different indices processing may cause deadlock
com.intellij.openapi.diagnostic.Logger$EmptyThrowable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:129)
	at com.intellij.util.indexing.IndexAccessValidator.checkAccessingIndexDuringOtherIndexProcessing(IndexAccessValidator.java:37)
	at com.intellij.util.indexing.FileBasedIndexImpl.processExceptions(FileBasedIndexImpl.java:829)
	at com.intellij.util.indexing.FileBasedIndexImpl.processValuesImpl(FileBasedIndexImpl.java:905)
	at com.intellij.util.indexing.FileBasedIndexImpl.getValues(FileBasedIndexImpl.java:771)
	at mobi.hsz.idea.gitignore.indexing.IgnoreFilesIndex.getEntries(IgnoreFilesIndex.java:233)
	at mobi.hsz.idea.gitignore.indexing.IgnoreFilesIndex.getFiles(IgnoreFilesIndex.java:249)
	at mobi.hsz.idea.gitignore.lang.kind.GitLanguage.getOuterFiles(GitLanguage.java:107)
	at mobi.hsz.idea.gitignore.indexing.ExternalIndexableSetContributor.getAdditionalFiles(ExternalIndexableSetContributor.java:77)
	at mobi.hsz.idea.gitignore.indexing.ExternalIndexableSetContributor.getAdditionalProjectRootsToIndex(ExternalIndexableSetContributor.java:102)
	at com.intellij.util.indexing.IndexableSetContributor.getProjectRootsToIndex(IndexableSetContributor.java:38)
	at com.intellij.util.indexing.FileBasedIndexImpl.collectScanRootRunnables(FileBasedIndexImpl.java:2258)
	at com.intellij.util.indexing.FileBasedIndexImpl.iterateIndexableFiles(FileBasedIndexImpl.java:2222)
	at com.intellij.util.indexing.FileBasedIndexImpl.projectIndexableFiles(FileBasedIndexImpl.java:1012)
	at com.intellij.util.indexing.FileBasedIndexImpl.lambda$processValuesImpl$2(FileBasedIndexImpl.java:883)
	at com.intellij.util.indexing.FileBasedIndexImpl.processExceptions(FileBasedIndexImpl.java:834)
	at com.intellij.util.indexing.FileBasedIndexImpl.processValuesImpl(FileBasedIndexImpl.java:905)
	at com.intellij.util.indexing.FileBasedIndexImpl.processValues(FileBasedIndexImpl.java:811)
	at com.intellij.util.indexing.FileBasedIndexImpl.processValues(FileBasedIndexImpl.java:801)
	at com.intellij.psi.search.FileTypeIndex.containsFileOfType(FileTypeIndex.java:133)
	at org.jetbrains.kotlin.idea.configuration.ConfigureKotlinInProjectUtilsKt$getModulesWithKotlinFiles$1.invoke(ConfigureKotlinInProjectUtils.kt:116)
	at org.jetbrains.kotlin.idea.configuration.ConfigureKotlinInProjectUtilsKt$getModulesWithKotlinFiles$1.invoke(ConfigureKotlinInProjectUtils.kt)
	at org.jetbrains.kotlin.idea.util.application.ApplicationUtilsKt$sam$Computable$ef13024b.compute(ApplicationUtils.kt)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:863)
	at org.jetbrains.kotlin.idea.util.application.ApplicationUtilsKt.runReadAction(ApplicationUtils.kt:24)
	at org.jetbrains.kotlin.idea.configuration.ConfigureKotlinInProjectUtilsKt.getModulesWithKotlinFiles(ConfigureKotlinInProjectUtils.kt:116)
	at org.jetbrains.kotlin.idea.configuration.ui.KotlinConfigurationCheckerComponent$projectOpened$1$1.run(KotlinConfigurationCheckerComponent.kt:71)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:307)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

### Steps to Reproduce

1. _First Step_ Update .ignore plugin on Android Studio to the latest version
2. _Second Step_ Restart Android Studio
3. _Third Step_ Open any project.

**Expected behavior:** _No errors._

**Actual behavior:** _Exception Error on .ignore plugin._

**Reproduces how often:** _At the start or at the opening of another project. Compiling a project doesn't trigger it._

### Versions

**Plugin:** 
```
 .ignore v2.2.0 (2017-09-04)
```

**IDE:** 
```
Android Studio 2.3.3
Build #AI-162.4069837, built on June 6, 2017
JRE: 1.8.0_112-release-b06 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
```

**OS:** 
```
Windows 10 Pro 
type x64
version 1703
comp. 15063.540
```

**Comment by nicity** [comment]
This issue causes deadlock in IntelliJ IDEA size: https://youtrack.jetbrains.com/issue/IDEA-178534

**Comment by hsz** [comment]
Can you please check if it's fixed with `v2.2.1-RC2`?

**Comment by Gerrit-K** [comment]
Sorry if this is the wrong place to ask, but is this issue related to [this community post](https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000563744-2017-4-hangs-during-indexing-)? I recently updated to 2017.2.4 and .ignore v.2.2.1 and my index keeps breaking every now and then. Indexing takes longer than usual and suddenly some files/classes are missing.
